### PR TITLE
[Storage] `az storage fs generate-sas`:  Support generate sas token for file system in ADLS Gen2 account

### PR DESCRIPTION
**Pull request by Juliehzl**
**Description**

Add `az storage fs generate-sas` support for ADLS gen2 account

**Testing Guide**

**History Notes**

[Component Name 1] BREAKING CHANGE: az command a: Make some customer-facing breaking change.
[Component Name 2] az command b: Add some customer-facing feature.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
Storage

File: src/azure-cli/azure/cli/command_modules/storage/_params.py [hunk 1]
```diff
@@ -1687,6 +1687,37 @@ def load_arguments(self, _):  # pylint: disable=too-many-locals, too-many-statem
...
+    with self.argument_context('storage fs generate-sas') as c:
+        from .completers import get_storage_acl_name_completion_list
+        t_file_system_permissions = self.get_sdk('_models#FileSystemSasPermissions',
+                                                 resource_type=ResourceType.DATA_STORAGE_FILEDATALAKE)
+        c.register_sas_arguments()
+        c.argument('file_system', options_list=['--name', '-n'], help="File system name.")
+        c.argument('id', options_list='--policy-name',
+                   help='The name of a stored access policy within the container\'s ACL.',
+                   completer=get_storage_acl_name_completion_list(t_container_permissions, 'container_name',
+                                                                  'get_container_acl'))
```
Comment (evelyn-ys) [inline src/azure-cli/azure/cli/command_modules/storage/_params.py:1699]: Oops, you forgot to change the copied code from container to fs🤣

Comment (Juliehzl) [reply src/azure-cli/azure/cli/command_modules/storage/_params.py:1699]: good catch

File: src/azure-cli/azure/cli/command_modules/storage/_validators.py [hunk 2]
```diff
@@ -754,12 +754,30 @@ def validate_metadata(namespace):
...
-    allowed_values = [x.lower() for x in dir(permission_class) if not x.startswith('__')]
+    allowed_values = get_permission_allowed_values(permission_class)
     return ' '.join(['({}){}'.format(x[0], x[1:]) for x in allowed_values])
 
 
+def get_permission_allowed_values(permission_class):
+    if permission_class:
+        instance = permission_class()
+
+        allowed_values = [x.lower() for x in dir(instance) if not x.startswith('_')]
```
Comment (evelyn-ys) [inline src/azure-cli/azure/cli/command_modules/storage/_validators.py:765]: Is it expected to change the filter from `not x.startswith('_')` to `not x.startswith('__')`? You do want to involve sth like `_abc`, right?

Comment (Juliehzl) [reply src/azure-cli/azure/cli/command_modules/storage/_validators.py:765]: Yes, it is expected. As we don not want to introduce `_abc`, we have to use `not x.startswith('_')`. If we use `not x.startswith('__')`, `_abc` with one dash will be involved.

**Review by evelyn-ys — APPROVED** [review]

**Pull request merged**